### PR TITLE
Run ID parameter added to RunQuery servlet 

### PR DESCRIPTION
**Pull request by jadecarino**
Signed-off-by: Jade Carino <jade.carino@ibm.com>

### CODE REVIEW

File: galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java [hunk 1]
```diff
@@ -46,295 +46,309 @@
...
+
+		/* Get list of Run Ids from the URL -
+		If a Run ID parameterlist is present in the URL then only return that run / those runs
+		Do not filter as well */
+
+		String runIdsParam = "";
+		if (paramMap.get("runId") != null && !paramMap.get("runId").isEmpty()) {
+			runIdsParam = paramMap.get("runId");
+			
+			String [] runIds = runIdsParam.split("[,]");
```
Comment (Michael-Baylis) [inline galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java:91]: wont just .split(",") do?

File: galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java [hunk 2]
```diff
@@ -46,295 +46,309 @@
...
+			for (String runId : runIds) {
+				try {
+					IRunResult run = getRunByRunId(runId);
```
Comment (Michael-Baylis) [inline galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java:94]: what happens if the runid is not found?

Comment (Michael-Baylis) [reply galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java:94]: ie,  if the runid is invalid or deleted,  it should be ignored and not added to the array

**Review by Michael-Baylis — CHANGES_REQUESTED** [review]

File: galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java (hunk 1 above)
Comment (jadecarino) [reply galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java:91]: yes that works, changed

File: galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java [hunk 3]
```diff
@@ -46,295 +46,314 @@
...
+			
+			String [] runIds = runIdsParam.split(",");
+			IRunResult run = null;
+			for (String runId : runIds) {
+				try {
+					run = getRunByRunId(runId.trim());
+					
+					if (run != null) {
+						runs.add(RunResultUtility.toRunResult(run, true));
+					} else {
```
Comment (Michael-Baylis) [inline galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunQuery.java:99]: dont need an empty else

**Review by Michael-Baylis — APPROVED** [review]

**Pull request merged**